**The main focus in this exercise is to explore any relationship between total tests completed and the dog**

We have looked at the relation in the group of each membership type, but in this exercise, we are gonna to look those data again in a whole sample scope.

In [1]:
import pandas as pd

df_user = pd.read_excel('/Users/jingai/workspace/Capstone-project/Dognition-coursera/raw-data/dognition_user.xlsx')
df_dog= pd.read_excel('/Users/jingai/workspace/Capstone-project/Dognition-coursera/raw-data/dognition_dog.xlsx')

In [2]:
print(df_user.columns)
print(df_user.shape)
print(df_dog.columns)
print(df_dog.shape)

Index(['Created_at', 'Updated_at', 'User_ID', 'Dog_ID', 'Test_name',
       'Subcategory_name', 'Gender', 'Birthday', 'Breed', 'Breed_Type',
       'Breed_Group', 'Weight', 'Dog_Fixed', 'DNA_Tested', 'Dimension',
       'Sign_in_Count', 'Max_Dogs', 'Membership_ID', 'Subscribed', 'City',
       'State', 'Zip', 'Country', 'Exclude', 'Free_Start_User',
       'Last_Active_At', 'Membership_Type', 'Rating', 'Rank_by_UserID',
       'Rank_by_DogID'],
      dtype='object')
(177481, 30)
Index(['Dog ID', 'Total Tests Completed', 'Mean ITI (days)',
       'Mean ITI (minutes)', 'Median ITI (days)', 'Median ITI (minutes)',
       'Time diff between first and last game (days)',
       'Time diff between first and last game (minutes)', 'User ID', 'Gender',
       'Birthday', 'Breed', 'Breed_Type', 'Breed_Group', 'Weight', 'Dog_Fixed',
       'DNA_Tested', 'Dimension', 'Sign_in_Count', 'Max_Dogs', 'Membership_ID',
       'Subscribed', 'City', 'State', 'Zip', 'Country', 'Exclude',
       'Free_Start_U

# 1 Total tests completed vs. dog info

## 1.1 Dog age

In [5]:
df_dog['Birthday'].head(5)

0    2011.0
1    2007.0
2    2012.0
3    2011.0
4    2010.0
Name: Birthday, dtype: float64

<font color = 'green'>scatter or a simple correlation calculation would help to check out the correlation between dog age and total tests completed</font>

## 2.1 Dog gender

In [9]:
df_dog['Gender'].value_counts()

In [13]:
#create a grouped dataframe based on the dog gender
gender_grp = df_dog.groupby(['Gender'])

#Check out the basic descriptive statistics in each gender group
gender_grp['Total Tests Completed'].apply(lambda x: x.describe().round(2))

Gender       
female  count    8633.00
        mean        9.85
        std         7.84
        min         1.00
        25%         4.00
        50%         7.00
        75%        16.00
        max        45.00
male    count    9352.00
        mean        9.77
        std         7.77
        min         1.00
        25%         4.00
        50%         7.00
        75%        16.00
        max        45.00
Name: Total Tests Completed, dtype: float64

### 2.1.1 No patterns between total tests completed and dog gender

Above output showed that female and male dog groups have quite similar distribution of total tests completed. So maybe grouping by the gender on its own does not generate any interesting patterns.


## 2.2 Dog breed info

### 2.2.1 Name of Breed

In [23]:
pd.unique(df_dog['Breed']).size

1327

It seems that there are 1327 unique breed names. Maybe we should start with breed group in that case.

### 2.2.2 Breed Group

In [24]:
df_dog['Breed_Group'].value_counts()

0               9055
Sporting        2587
Herding         1876
Toy             1168
Non-Sporting    1007
Working          899
Terrier          809
Hound            584
Name: Breed_Group, dtype: int64

In [32]:
#group by dog data by breed group

breed_grp = df_dog.groupby(['Breed_Group'])
breed_grp['Total Tests Completed'].apply(lambda x: x.median().round(1))

Breed_Group
0               7.0
Herding         7.0
Hound           7.0
Non-Sporting    7.0
Sporting        7.0
Terrier         7.0
Toy             5.0
Working         7.0
Name: Total Tests Completed, dtype: float64

### 2.2.3 Percetage of dogs completed 20 tests in each breed group

In [41]:
#Select dogs completed the first 20 tests only
filt_tests_comp = (df_dog['Total Tests Completed'] >= 20)
df_dog_comp = df_dog.loc[filt_tests_comp]

#Count dogs who completed the first 20 tests in each breed group
comp_breed_grp = df_dog_comp.groupby(['Breed_Group'])
comp_count = comp_breed_grp['Total Tests Completed'].apply(lambda x: x.count())

comp_count

Breed_Group
0               2059
Herding          523
Hound            135
Non-Sporting     217
Sporting         675
Terrier          172
Toy              206
Working          212
Name: Total Tests Completed, dtype: int64

In [43]:
breed_counts = df_dog['Breed_Group'].value_counts()
breed_counts

0               9055
Sporting        2587
Herding         1876
Toy             1168
Non-Sporting    1007
Working          899
Terrier          809
Hound            584
Name: Breed_Group, dtype: int64

In [55]:
perc_tests_comp = ((comp_count/breed_counts)*100).round()

df_breed_grp = pd.concat([breed_counts, comp_count, perc_tests_comp], axis=1)
df_breed_grp.rename(columns={'Breed_Group':'NumBreed','Total Tests Completed':'NumTestsComp', 0:'PercComp'}, inplace = True)
df_breed_grp=df_breed_grp.sort_values(by='PercComp', ascending=False)

df_breed_grp

NumBreed  NumTestsComp  PercComp
Herding           1876           523      28.0
Sporting          2587           675      26.0
Working            899           212      24.0
0                 9055          2059      23.0
Hound              584           135      23.0
Non-Sporting      1007           217      22.0
Terrier            809           172      21.0
Toy               1168           206      18.0

#### 2.2.3.1 First findings

* Herding dogs tend to complete tests 